src/combat.cpp

```diff
@@ -516,9 +530,13 @@ void Combat::CombatManaFunc(Creature* caster, Creature* target, const CombatPara
 {
 	assert(data);
 	CombatDamage damage = *data;
-	if (damage.primary.value < 0) {
-		if (caster && caster->getPlayer() && target->getPlayer()) {
-			damage.primary.value /= 2;
+	if ((damage.primary.value < 0 || damage.secondary.value < 0) && caster) {
```

Comment:
I tested and manadrain spells used by monsters caused server crash. With 
```suggestion
	if ((damage.primary.value < 0 || damage.secondary.value < 0)) {
		if (caster && caster->getPlayer() && target->getPlayer()) {
```
as before it seems to work fine.